### ISSUE TITLE
Calling `dotnet tool restore --verbosity quiet` shouldn't output anything on success

### ISSUE THREAD

**Issue by tompazourek** [dotnet/sdk#16535]
In various PS scripts, I often call `dotnet tool restore` in order to ensure that the dotnet tools I plan to call are actually restored.

The output is:

```
Tool 'xxx' (version '1.1.1') was restored. Available commands: xxx
Tool 'yyy' (version '2.2.2') was restored. Available commands: yyy
Tool 'zzz' (version '3.3.3') was restored. Available commands: zzz

Restore was successful.
```

I need to display this output because sometimes it contains stuff like `https://microsoft.com/devicelogin` instructions and code.

But in most cases, I'd prefer that the `dotnet tool restore` was more quiet.

I found out (by calling `--help`) that there's a `--verbosity` switch.

However, even when I call `dotnet tool restore --verbosity quiet`, it still outputs the same amount of data. It seems that the verbosity switch has no effect here.

Would it be possible to have a functioning `--verbosity quiet`? I'd expect that it wouldn't output anything if everything is successful, and it would only output something in case something went wrong or it needs some action. Or maybe that's what `--verbosity minimal` is for?

**Comment by jeremy-visionaid** [comment]
Yeah, this is pretty annoying for cake runners (e.g. build.ps1). I'd expect the restore to be quiet, and the verbosity argument working would be much nicer than redirection

**Comment by kieronlanning** [comment]
@marcpopMSFT I see you added this to the backlog - is a discusson/ way of bumping the priority possible? 

It's not a good experience for automations.

For example, using dotnet format with a pre-commit hook causes VS2026 to think the commit failed because of the 'restoring tool xxx' output.

**Comment by marcpopMSFT** [comment]
Triage: since the .NET tools rewrite, verbosity does apply to the NuGet output. 

Are you saying just suppress all messages on the restore command if using quiet? Should there be any output at all if it should succeed? We could potentially just do that for quiet only.

**Comment by kieronlanning** [comment]
Hi! I think that's correct - zero output on `-q/ --quiet` unless there's a failure?

Assuming exit codes will be appropriate too?

**Comment by marcpopMSFT** [comment]
Triage: See if copilot can do the above where quiet for tool restore removes all output to STDOUT but the exit code is still accurate. baronfel says that any error could still go to STDERR.
